libgnucash/engine/qof-string-cache.cpp

```diff
@@ -62,48 +62,38 @@ qof_string_cache_clear (void)
 void
 qof_string_cache_remove(const char * key)
 {
-    if (key && *key)
-    {
-        auto cache_iter = qof_string_cache.find (key);
+    if (!key || !key[0])
+        return;
 
-        if (cache_iter == qof_string_cache.end())
-            return;
+    auto cache_iter = qof_string_cache.find (key);
+    if (cache_iter == qof_string_cache.end())
+        return;
 
-        auto& refcount = cache_iter->second;
-
-        if (refcount == 1)
-            qof_string_cache.erase (cache_iter);
-        else
-            refcount--;
-    }
+    auto& refcount = cache_iter->second;
+    if (refcount == 1)
+        qof_string_cache.erase (cache_iter);
+    else
+        refcount--;
 }
```

Comment:
I was looking for algorithms in the Algorithm library or member functions of an unordered map to reduce the number of `if` clauses, but in this case I didn't find much that's useful. The only possible change I'd consider to make the code slightly more in line with my suggestion for insert is to first decrease the value by one and then check if it has gone to 0 to delete it.

```
(cache_iter->second)--;
if (!cache_iter->second)
        qof_string_cache.erase (cache_iter);
```